azurelinuxagent/common/rdma.py

```diff
@@ -110,6 +109,26 @@ def get_rdma_version():
             logger.error(error_msg % driver_info_source)
             return
 
+    @staticmethod
+    def is_kvp_daemon_running():
+        """Look for kvp daemon names in ps -ef output and return True/False
+        """
+        # for centos, the hypervkvpd and the hv_kvp_daemon both are ok.
+        # for suse, it uses hv_kvp_daemon
+        kvp_daemon_names = ['hypervkvpd', 'hv_kvp_daemon']
+
+        exitcode, ps_out  = shellutil.run_get_output("ps -ef")
+        if exitcode != 0:
+            raise Exception('RDMA: ps -ef failed: %s' % ps_out)
+        for n in  kvp_daemon_names:
+            if n in ps_out:
+                logger.info('RDMA: kvp daemon (%s) is running' % n)
+                return True
+            else:
+                logger.verbose('RDMA: kvp daemon (%s) is not running' % n)
+        return False
+
```

Comment:
Tying this to RDMA creates, IMHO, a very tight coupling between RDMA and something that appears to be generic. The kvp daemon doe not only exist to provide RDMA information, I think. It appears to me that this code should live somewhere else. Given that there is work going on for PCI pass through I would expect that the kvp daemon will also be involved there. It feels like this code should live in code located in azurelinuxagent/common/utils/.

Also there is another option for generalization in that the call that gets the process list and error checking should probably live in a method called get_running_processes() such that it can be reused for other things we will be looking for.